haskellPackages: update stackage and hackage

maralorn opened the pull request:
### This Merge

This PR is the regular merge of the `haskell-updates` branch into `master`.

This branch is being continually built and tested by hydra at https://hydra.nixos.org/jobset/nixpkgs/haskell-updates. You may be able to find an up-to-date Hydra build report at [cdepillabout/nix-haskell-updates-status](https://github.com/cdepillabout/nix-haskell-updates-status).

We roughly aim to merge these `haskell-updates` PRs at least once every two weeks. See the @NixOS/haskell [team calendar](https://cloud.maralorn.de/apps/calendar/p/Mw5WLnzsP7fC4Zky) for who is currently in charge of this branch.

### haskellPackages Workflow Summary

Our workflow is currently described in [`pkgs/development/haskell-modules/HACKING.md`](https://github.com/NixOS/nixpkgs/blob/haskell-updates/pkgs/development/haskell-modules/HACKING.md).

The short version is this:
* We regularly update the Stackage and Hackage pins on `haskell-updates` (normally at the beginning of a merge window).
* The community fixes builds of Haskell packages on that branch.
* We aim at at least one merge of `haskell-updates` into `master` every two weeks.
* We only do the merge if the [`mergeable`](https://hydra.nixos.org/job/nixpkgs/haskell-updates/mergeable) job is succeeding on hydra.
* If a [`maintained`](https://hydra.nixos.org/job/nixpkgs/haskell-updates/maintained) package is still broken at the time of merge, we will only merge if the maintainer has been pinged 7 days in advance. (If you care about a Haskell package, become a maintainer!)

---

This is the follow-up to #157026. Come to [#haskell:nixos.org](https://matrix.to/#/#haskell:nixos.org) if you have any questions.

maralorn committed:
all-cabal-hashes: 2022-01-27T16:52:24Z -> 2022-01-31T19:23:21Z

This commit has been generated by maintainers/scripts/haskell/update-hackage.sh

maralorn committed:
haskellPackages: regenerate package set based on current config

This commit has been generated by maintainers/scripts/haskell/regenerate-hackage-packages.sh

sternenseemann committed:
haskellPackages.tz: drop upstreamed patch

sternenseemann committed:
haskellPackages.nix-tree: reflect brick update

github-actions[bot] committed:
Merge master into haskell-updates

vojta001 committed:
haskellPackages.contiguous: unmark as broken

It currently builds fine without any modifications.

sternenseemann committed:
haskellPackages.git-annex: adjust src hash for 10.20220127

sternenseemann commented:
- [ ] Get this to work or install the file(s) manually

```
./Build/InstallDesktopFile /nix/store/zd9x9clipjhjpabymgdbff7g3g7n2i9p-git-annex-10.20220127/bin/git-annex || true
InstallDesktopFile: /usr: createDirectory: permission denied (Permission denied)
```

github-actions[bot] committed:
Merge master into haskell-updates

cocreature committed:
haskell.compiler.ghc902: Switch to LLVM 12

Currently, the build on MacOS ARM64 is broken because LLVM 9 (or more
specifically compiler-rt) is broken (and is marked broken). Both
8.10.7 and 9.2.1 are already set to LLVM 12 so this PR adjusts this to
also use LLVM 12 for GHC 9.0.2 which seems to get things building for me.

github-actions[bot] committed:
Merge master into haskell-updates

akshaymankar committed:
haskell.packages.ghc901.generic-lens*: 2.2.0.0 -> 2.2.1.0

hackage-packages.nix doesn't have generic-lens_2_2_0_0, perhaps this got missed
in the last update.

github-actions[bot] committed:
Merge master into haskell-updates

sternenseemann committed:
haskell.packages.ghc901.primitive-extras: 0.10.1 -> 0.10.1.4

This update resolves a compilation failure we were seeing with 9.0.1.

sternenseemann committed:
haskell.packages.ghc921.haskell-src-meta: drop obsolete patch

sternenseemann committed:
haskell.packages.ghc921.ghc-exactprint: provide missing data-default

github-actions[bot] committed:
Merge master into haskell-updates

maralorn committed:
haskellPackages.haskell-language-server: Fix dependencies and refactor overrides

maralorn committed:
haskell.packages.ghc902.haskell-language-server: Enable all working plugins

maralorn committed:
haskell.packages.ghc921.haskell-language-server: Fix build and enable

maralorn committed:
Merge pull request #158192 from maralorn/hls-1.6

[haskell-updates] haskell-language-server: 1.5 -> 1.6

github-actions[bot] committed:
Merge master into haskell-updates

sternenseemann committed:
haskell-language-server: disable GHC 9.2.1 by default on aarch64

GHC 9.2.1 is still very much broken on aarch64, unfortunately.

maralorn committed:
haskell-language-server: Disable flaky tests for some plugins

siraben committed:
Merge pull request #158347 from sternenseemann/hls-disable-9.2.1-aarch64

[haskell-updates] haskell-language-server: disable GHC 9.2.1 by default on aarch64

maralorn committed:
Merge branch 'master' into haskell-updates

maralorn committed:
haskellPackages: mark builds failing on hydra as broken

This commit has been generated by maintainers/scripts/haskell/mark-broken.sh

maralorn committed:
haskellPackages: mark builds failing on hydra as broken

This commit has been generated by maintainers/scripts/haskell/mark-broken.sh
